01_fib/fibonacci.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func calculateFib(n int) []int {
+	f0, f1, result := 1, 1, []int{}
+
+	for i := 1; i <= n; i++ {
+		result = append(result, f0)
+		f0, f1 = f1, f0+f1
+	}
+	return result
+}
+
+func fib(n int) {
+	//ideally this can be printed in main but lab needs to be printed inside the func
```

Comment:
I roughly agree! A middle ground might be to name this function something like "formatSequence" with the signature []int -> string , i.e. a function that takes an input sequence of numbers and produces a formatted string .  Then main could do something like  `fmt.Fprint(os.Stdout, formatSequence(calculateFib(n)))`

Reply:
fixed this, thank you